### ISSUE TITLE
Named Parameters with template strings collision

### ISSUE THREAD

**Issue by pkoretic** [vitaly-t/pg-promise#24]
since named parameters have the same es6 template string syntax for variables, there is a collision if you want to use both

this will throw error
```
db.query(`select * from users where name=${name} and active=${active}`, {
    name: 'John',
    active: true
});
```
one reason why would you want to use template strings in this case is for multiline support without concating

so you can write for example
``` 
`
select * from users
where name = ${name}
and active = ${active}
`
```
instead of 

```
"select * from users" +
"where name = ${name}" +
"and active = ${active}" 
```

**Comment by vitaly-t** [comment]
ES6 specification uses back-tick symbols for text, to avoid conflicts with regular text wrapped in single or double-quotes.

So, how exactly can you get named parameters to produce a conflict? In your example I don't use quotes at all, which is invalid in any syntax standard.

**Comment by pkoretic** [comment]
var test = "test"
console.log(`this is a ${test}`)

this is one of the features of es6 template strings, so you can see why using 'your' syntax makes a problem, since template strings expect that variable to exists, but you use the same syntax and actually bind the variable later.
So when you declare your sql using template strings and your named parameters es6 template strings fail because there is no variable by that name that es6 syntax expects.

**Comment by vitaly-t** [comment]
It's actually like this:
```javascript
var test = "test";
console.log(`this is a ${test}`);
```

**Comment by pkoretic** [comment]
@vitaly-t yeah, fixed it in the meantime ;)

**Comment by vitaly-t** [comment]
Do I understand you correctly what you are saying? - 
You want to join ES6 syntax with pg-promise named variables just for the sake of multi-line support?

**Comment by pkoretic** [comment]
@vitaly-t I don't think that is relevant. You shouldn't use standard language keywords for custom functionality which obviously collides. This is why sequelize for example use '?' and ':' for position and named binds.

**Comment by vitaly-t** [comment]
Perhaps, but you are talking about a very new standard here. And if it really becomes an issue, it would be very easy to fix. I can mark it as a feature request.

**Comment by pkoretic** [comment]
@vitaly-t  template strings are in the release candidate stage implemented in all major software http://kangax.github.io/compat-table/es6/.   
It's better to avoid it sooner than later since this is not backward compatibility change for your library.

**Comment by vitaly-t** [comment]
It can be made backward compatible, through an extra option to override the prefix symbol explicitely with one that's ok for ES6, like `?` or `:` - will have to decide.

**Comment by vitaly-t** [comment]
Anyway, let's consider this an improvement from now on. Feel free throwing the best way to change it, in terms of the syntax. I will try to get around implementing it this weekend, if we make a decision.

**Comment by pkoretic** [comment]
Ok, in general whatever suits you best. I'm for one that's fastest to parse so I'll do some tests :) Thanks.

**Comment by vitaly-t** [comment]
The change will be only in the front symbol, from `$` to something else, so the parsing speed and logic will be exactly the same. I just need to figure out what would be the best alternative symbol to use :)

`%&?:*`

Alternatively, I could change it to `$()` what do you reckon? ;)

**Comment by pkoretic** [comment]
Ok, $(var) of maybe even ?{var}. Seems easiest to write (on english keyboard at least)

**Comment by vitaly-t** [comment]
I believe i can easily add support for $() in addition, as opposed to instead of. In fact, i think this will be the best solution, for all means and purposes.

**Comment by pkoretic** [comment]
Ok, looks great.

**Comment by vitaly-t** [comment]
I have already checked in all the changes to support both `${}` syntax and `$()` at the same time. You can see that the change was minimal.

There will be a few updates in documentation and tests though before I do a release with it. Sometime before Monday, I hope.

**Comment by pkoretic** [comment]
Looks great! We will then switch to using it from monday if released.

**Comment by vitaly-t** [comment]
This issue has been resolved in the new [1.2.0 release](https://github.com/vitaly-t/pg-promise/releases/tag/v.1.2.0).

**Comment by vitaly-t** [comment]
@pkoretic you may like the syntax update in [release 1.3.1](https://github.com/vitaly-t/pg-promise/releases/tag/v.1.3.1) ;)
